### PR TITLE
Add Kokkos profiling regions

### PR DESCRIPTION
**Pull request by streeve**
Focus on parallel iteration and communication first; should also add to main algorithms

### CODE REVIEW

File: core/src/Cabana_Parallel.hpp [hunk 1]
```diff
@@ -249,6 +253,8 @@ inline void neighbor_parallel_for(
     const FunctorType& functor, const NeighborListType& list,
     const FirstNeighborsTag, const SerialOpTag, const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
Comment (dalg24) [inline core/src/Cabana_Parallel.hpp:256]: Are these `neighbor_parallel_for ` regions really useful?  These seem to only a single kernel launch with nothing other than construct the policy.  Did you look at some profiles that made you think that we should add them?
A possible alternative would be to give a default value not empty to the `str` argument.  Presumably when provided by the user that label is good enough and we don't really need the region.

Comment (streeve) [reply core/src/Cabana_Parallel.hpp:256]: That's a fair question. It is more useful for the Cabana algorithms/communication, but I do like showing where Cabana is being used (even when it effectively duplicates the user kernel in the profile)

**Comment by abisner** [comment]
Can we add regions for `Cabana::binByKey`?

**Review by abisner — COMMENTED** [review]
Can we add regions for `Cabana::binByKey()`? It is used downstream.

**Comment by streeve** [comment]
@abisner added for `binByKey`. I left the (simd/neighbor/grid) parallel regions for now, but can switch to appending-to or replacing-empty parallel labels if you prefer as well

File: core/src/Cabana_Parallel.hpp [hunk 2]
```diff
@@ -684,6 +712,8 @@ inline void neighbor_parallel_reduce(
     const SecondNeighborsTag, const SerialOpTag, ReduceType& reduce_val,
     const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
Comment (abisner) [inline core/src/Cabana_Parallel.hpp:715]: `Cabana::neighbor_parallel_reduce`

File: core/src/Cabana_Parallel.hpp [hunk 3]
```diff
@@ -757,6 +789,8 @@ inline void neighbor_parallel_reduce(
     const FirstNeighborsTag, const TeamOpTag, ReduceType& reduce_val,
     const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
Comment (abisner) [inline core/src/Cabana_Parallel.hpp:792]: `Cabana::neighbor_parallel_reduce`

File: core/src/Cabana_Parallel.hpp [hunk 4]
```diff
@@ -834,6 +870,8 @@ inline void neighbor_parallel_reduce(
     const SecondNeighborsTag, const TeamOpTag, ReduceType& reduce_val,
     const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
Comment (abisner) [inline core/src/Cabana_Parallel.hpp:873]: `Cabana::neighbor_parallel_reduce`

File: core/src/Cabana_Parallel.hpp [hunk 5]
```diff
@@ -916,6 +956,8 @@ inline void neighbor_parallel_reduce(
     const SecondNeighborsTag, const TeamVectorOpTag, ReduceType& reduce_val,
     const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
Comment (abisner) [inline core/src/Cabana_Parallel.hpp:959]: `Cabana::neighbor_parallel_reduce`

**Review by abisner — COMMENTED** [review]
A few naming typos in Cabana_Parallel regions.

File: core/src/Cabana_Parallel.hpp [hunk 6]
```diff
@@ -618,6 +642,8 @@ inline void neighbor_parallel_reduce(
     const FirstNeighborsTag, const SerialOpTag, ReduceType& reduce_val,
     const std::string& str = "" )
 {
+    Kokkos::Profiling::pushRegion( "Cabana::neighbor_parallel_for" );
```
[the remark this answers is not in the document]
Comment (streeve) [reply core/src/Cabana_Parallel.hpp]: fixed

**Review by abisner — APPROVED** [review]

**Pull request merged**